### PR TITLE
Remove extra warning alert from SmtpSettings.Edit

### LINKED ISSUE #14286: Two alerts displayed in admin settings to reload the current tenant
The `/Admin/Settings/email` page displays 2 identical warning alerts:
`The current tenant will be reloaded when the settings are saved.`

image

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #14286

### CODE REVIEW

**Comment by hishamco** [comment]
I noticed this happened after the changes in https://github.com/OrchardCMS/OrchardCore/pull/14150 , so there are some other places that we need to fix

Before that, I'd like to ask @MikeAlhayek if he has an idea to override the alert - if it's required - when he did this change before

**Comment by agriffard** [comment]
Is there the same on /admin/settings/localisation? Can you please fix it too?

**Comment by hishamco** [comment]
Yes, there are already in my local, but I need @MikeAlhayek feedback coz some need tweaks

**Comment by hishamco** [comment]
The once that need to override are related to AUTH modules

**Comment by MikeAlhayek** [comment]
@hishamco I think the problem is happening because the `Settings_Wrapper__General` wrapper is injected every time. You should undo all your changes and try adding the following code before https://github.com/OrchardCMS/OrchardCore/blob/033e0ae1f5e1ed421b5e919082be2fe40d4f2c53/src/OrchardCore.Modules/OrchardCore.Settings/Drivers/DefaultSiteSettingsDisplayDriver.cs#L33

```
if (!context.GroupId.Equals(GroupId, StringComparison.OrdinalIgnoreCase))
{
    return Task.FromResult<IDisplayResult>(null);
}
```

To clean up `DefaultSiteSettingsDisplayDriver` a bit more, I would even add a new method like this
```
private bool IsGeneralSettings() => context.GroupId.Equals(GroupId, StringComparison.OrdinalIgnoreCase)
```
Then use `IsGeneralSettings()` on line 33 and in line 52. 

On line 52, I would even save indentation by replacing like 52 with this
```
if (!IsGeneralSettings())
{
    return null;
}
```

**Comment by jtkech** [comment]
Maybe to make it simpler, see https://github.com/OrchardCMS/OrchardCore/issues/14286#issuecomment-1714659693

File: src/OrchardCore.Modules/OrchardCore.Settings/Drivers/DefaultSiteSettingsDisplayDriver.cs [hunk 1]
```diff
@@ -49,7 +54,7 @@ public override Task<IDisplayResult> EditAsync(ISite site, BuildEditorContext co
 
         public override async Task<IDisplayResult> UpdateAsync(ISite site, UpdateEditorContext context)
         {
-            if (context.GroupId.Equals(GroupId, StringComparison.OrdinalIgnoreCase))
+            if (IsGeneralSettings(context))
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Settings/Drivers/DefaultSiteSettingsDisplayDriver.cs:57]: Can you please save dentition here by adding a return when not group? Look what I did in my PR.

File: src/OrchardCore.Modules/OrchardCore.Settings/Drivers/DefaultSiteSettingsDisplayDriver.cs [hunk 2]
```diff
@@ -104,5 +109,8 @@ private static void PopulateProperties(ISite site, SiteSettingsViewModel model)
             model.AppendVersion = site.AppendVersion;
             model.CacheMode = site.CacheMode;
         }
+
+        private static bool IsGeneralSettings(BuildEditorContext context)
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Settings/Drivers/DefaultSiteSettingsDisplayDriver.cs:113]: @hishamco 
Maybe rename this to `IsGeneralGroup`. Ping me once done and I'll merge it

**Review by MikeAlhayek — CHANGES_REQUESTED** [review]

**Review by MikeAlhayek — APPROVED** [review]

**Pull request merged**